hooks/useCustomStyle.tsx

```diff
@@ -0,0 +1,122 @@
+    const mainLeftPadding = getOneSidePadding(mainElement)
+
+    const allowedContentFullWidth = screenWidth - (mainLeftPadding * 2)
+
+    const elementXPos = currentElement.getBoundingClientRect().x - mainLeftPadding
+    const elementXPosRatio = elementXPos / allowedContentFullWidth
+
+    return {
+        elementXPos,
+        allowedContentFullWidth,
+        elementXPosRatio,
+    }
+}
+
+function computeTwoSlashOverrideStyle({isMounted,
+                                        contentWidth,
+                                        allowedContentFullWidth,
+                                        elementXPos,
+                                      } :
+                                      {isMounted: boolean,
+                                      contentWidth: number ,
+                                      allowedContentFullWidth: number ,
+                                      elementXPos: number ,
+                                      }) {
+
+    return  isMounted
+            ?(
+                contentWidth >= allowedContentFullWidth
+                ? `pre.twoslash data-lsp:hover::before {
+                left: 0px !important;
+                }`
+                : elementXPos + contentWidth >= allowedContentFullWidth
+                ? `pre.twoslash data-lsp:hover::before {
+                left: ${allowedContentFullWidth - contentWidth}px !important;
+                }`
+                : `pre.twoslash data-lsp:hover::before {
+                left: ${elementXPos}px !important;
+                }`
+            )
+            : "";
+}
+
+export function useCustomStyle() {
+
+  const [isMounted, setIsMounted] = useState(false);
+  const [currentStyle, setCurrentStyle] = useState("");
+
+  function rePositionTwoSlashBox(e:Event) {
+
+    const currentElement = e.target as (HTMLElement | null)
+    if( currentElement == null || !isMobile()) return
+    if(isInCodeContainer(currentElement) == null) return
+
+    const {
+            elementXPos,
+            allowedContentFullWidth,
+        } =  getValuesForCSSLeft(currentElement)
+
+    const contentWidth = getPseudoElementWidth(currentElement, ':before');
+
+    setCurrentStyle(
+        computeTwoSlashOverrideStyle({
+            isMounted,
+            contentWidth,
+            allowedContentFullWidth,
+            elementXPos,
+        })
+    )
+  }
+
+  useEffect(() => {
+      setIsMounted(true);
+      return () => setIsMounted(false);
+  },[])
+
+  const touchAndClickListener = (e : Event) => {
+      e.preventDefault()
+      rePositionTwoSlashBox(e)
+  }
+
+  useEffect(() => {
+      if(isMounted ) {
+          window.addEventListener("touch", touchAndClickListener)
+          window.addEventListener("click", touchAndClickListener)
+      }
+
+      return () => {
+          window.removeEventListener("touch", touchAndClickListener)
+          window.removeEventListener("click", touchAndClickListener)
+      }
+  }, [isMounted]);
```

Comment:
`useEffect` will always run a _after_ a component has mounted, meaning that `isMounted` actually checks for the **2nd** render of a component and not the mount of the component.

This effect violates the rules of hooks, `touchAndClickListener` is _not_ a static or a stable function, therefore it should be in the deps array.
But more likely the flow control here need to change in general to move the function inside the `useEffect`

Reply:
removed "isMounted" from Dependency Array

removed "if(isMounted)"

"touchAndClickListener function" moved inside the useEffect